include/zephyr/zbus/zbus.h

```diff
@@ -717,6 +724,89 @@ static inline void zbus_chan_set_msg_sub_pool(const struct zbus_channel *chan,
 
 #endif /* ZBUS_MSG_SUBSCRIBER_NET_BUF_POOL_ISOLATION */
 
+#if defined(CONFIG_ZBUS_CHANNEL_PUBLISH_STATS) || defined(__DOXYGEN__)
+
+/**
+ * @brief Update the publishing statistics for a channel
+ *
+ * This function updates the publishing statistics for the @ref zbus_chan_claim ->
+ * @ref zbus_chan_finish workflow, which cannot automatically determine whether
+ * new data has been published or not.
+ *
+ * @warning This function must only be used directly for already locked channels.
+ *
+ * @param chan The channel's reference.
+ */
+static inline void zbus_chan_pub_stats_update(const struct zbus_channel *chan)
+{
+	__ASSERT(chan != NULL, "chan is required");
+
+	chan->data->publish_timestamp = k_uptime_ticks();
+	chan->data->publish_count += 1;
+}
+
+/**
+ * @brief Get the time a channel was last published to.
+ *
+ * @note Will return 0 if channel has not yet been published to.
+ *
+ * @param chan The channel's reference.
+ *
+ * @return The kernel timestamp of the last publishing action.
+ */
+static inline k_ticks_t zbus_chan_pub_stats_last_time(const struct zbus_channel *chan)
+{
+	__ASSERT(chan != NULL, "chan is required");
+
+	return chan->data->publish_timestamp;
+}
+
+/**
+ * @brief Get the number of times a channel has been published to.
+ *
+ * @note Will return 0 if channel has not yet been published to.
+ *
+ * @param chan The channel's reference.
+ *
+ * @return The number of times a channel has been published to.
+ */
+static inline uint32_t zbus_chan_pub_stats_count(const struct zbus_channel *chan)
+{
+	__ASSERT(chan != NULL, "chan is required");
```

Comment:
Please use `_ZBUS_ASSERT` for the functions. On the tests, please add a test that checks the result is `-EFAULT` the the `chan` is NULL. I followed the specification-based tests, and we must ensure all the bad values were tested. 

```suggestion
	_ZBUS_ASSERT(chan != NULL, "chan is required");
```

Reply:
What's the reason for shifting to `_ZBUS_ASSERT`? Every other inline function in this header uses `__ASSERT`.

Reply:
Take the `zbus_obs_is_enabled` as a good example. I added myself mistakenly the other inconsistencies (`__ASSERT` instead of `_ZBUS_ASSERT`). I will send a PR fixing that. Please follow the path by testing the "bad" values. Okay? Thanks for pointing out the inconsistency.

Reply:
I'm sorry. I'm a bit confused. There are too many things in my head now, and I am trying to be fast on the review but missing the details. 

That makes sense for the functions you are mentioning. But the rest of the code, the functions that return an error code, where we are testing whether the wrong value is being verified even with an assertion, is important. If someone removes the verification in the future, the automated tests will not detect code quality degradation. Verifying the function's arguments is necessary to avoid obscure errors while using the subsystem.

Reply:
The tests must check the invalid arguments, which I did for all zbus functions.

Reply:
> That makes sense for the functions you are mentioning.

These 4 functions are the only functions in this PR.

> The tests must check the invalid arguments, which I did for all zbus functions.

Take `zbus_chan_pub_stats_update` as an example. In any application that is not the ZBUS testing applications, there are only 3 possible things that can happen.
* `channel != NULL`: function works as intended, nothing to return
* `channel == NULL`, `CONFIG_ASSERT=y`: assert logic runs, function never returns
* `channel == NULL`, `CONFIG_ASSERT=n`: code tries to dereference null, exception triggers, function never returns

If someone deletes the `__ASSERT` check, you are just in the 3rd situation again.

Lets pretend that `__ASSERT` is swapped for `_ZBUS_ASSERT` and `void` is swapped for `int` as the return value.
Considering the above 3 situations, there are no situations in which the function can return a value other than 0.
But because the return type is not `void`, defensive programming would require checking the return code everywhere.
So this is just adding a pile of boilerplate to every single user of this API without any benefit.

The other 3 functions can have the same reasoning applied.

Reply:
This is true for the functions you are adding. I have already approved the PR. Thanks for the contribution. Sorry for the confusion. It's great to see others contributing to zbus.